mixins/test/localize-mixin.test.js

```diff
@@ -297,6 +329,104 @@ describe('LocalizeMixin', () => {
 
 	});
 
+	describe('localizeHTML', async() => {
+
+		const elem = await fixture(`<${localizeHTMLTag}></${localizeHTMLTag}>`);
+
+		const getRenderString = data => {
+			if (data.constructor === String || data.constructor === Number) return data;
+			if (Array.isArray(data)) {
+				return data.map(e => getRenderString(e)).join('');
+			}
+			const { strings, values } = data;
+
+			if (!strings || !values) return data;
+
+			const flatValues = [...values, ''].map(v => {
+				if (typeof v === 'object') return getRenderString(v);
+				if (v.toString() === 'Symbol(lit-nothing)') {
+					return '{lit-nothing}';
+				}
+				return v;
+			});
+			return strings.reduce((acc, s, idx) => acc + s + flatValues[idx], '').replace(/\s[\w-]+="{lit-nothing}"/g, '');
+		};
+
+		it('should replace acceptable markup with correct HTML', async() => {
+			const defaultTags = elem.localizeHTML('test1');
+			const manual = elem.localizeHTML('test2', { link: chunks => localizeMarkup`<d2l-link href="http://d2l.com">${chunks}</d2l-link>` });
+			const disallowed = elem.localizeHTML('test3', { link: chunks => localizeMarkup`<div>${chunks}</div>` });
+			const badTemplate = elem.localizeHTML('test3', { link: chunks => html`${chunks}` });
+			const tooltip = elem.localizeHTML('test4', { tooltip: generateTooltipHelp({ contents: 'Tooltip text' }) });
+
+			const items = ['milk'];
+			const pluralLink = elem.localizeHTML('pluralTest', { itemCount: items.length, item: items[0], link: generateLink({ href: 'checkout' }) });
+
+			items.push('bread', 'eggs');
+			const pluralMap = elem.localizeHTML('pluralTest', { itemCount: items.length, link: generateLink({ href: 'checkout' }), html: () => items.map(i => localizeMarkup`<p>${i}</p>`) });
+
+			expect(getRenderString(defaultTags)).to.equal('This is <strong>important</strong>, this is <strong><em>very important</em></strong>');
+			expect(getRenderString(manual)).to.equal('This is <d2l-link href="http://d2l.com">a link</d2l-link>');
+
+			expect(getRenderString(disallowed)).to.equal('This is <link>replaceable</link>');
+			expect(getRenderString(badTemplate)).to.equal('This is replaceable');
+			expect(getRenderString(tooltip)).to.equal('This is a <d2l-tooltip-help inherit-font-style text="tooltip-help">Tooltip text</d2l-tooltip-help> within a sentence');
+			expect(getRenderString(pluralLink)).to.equal('You have milk in your cart. <d2l-link href="checkout">Checkout</d2l-link>');
+			expect(getRenderString(pluralMap)).to.equal('Items in your cart:<p>milk</p><p>bread</p><p>eggs</p><d2l-link href="checkout">Checkout</d2l-link>');
+		});
+
+		let a;
+		[{
+			get renderString() { return elem.localizeHTML('typeChecker', { a }); },
```

Comment:
I guess this should just be `undefined` but it hurts.
```suggestion
		[{
			get renderString() { return elem.localizeHTML('typeChecker', { a: undefined }); },
```

Reply:
Haha yep, I'd say so. I'm curious about your reluctance towards the [`undefined` primitive](https://developer.mozilla.org/en-US/docs/Glossary/Undefined) actually, that's a new one to me.

Reply:
Basically just the [eslint writeup](https://eslint.org/docs/latest/rules/no-undefined) on it. Just one of those things I've avoided for years because it's fairly easy to do usually.

Reply:
Neat! I didn't know undefined could be overridden, I assumed it was equivalent to `null`.